workbench-core/authorization/src/caslAuthorizationPlugin.ts

```diff
@@ -38,4 +69,30 @@ export default class CASLAuthorizationPlugin implements AuthorizationPlugin {
     });
     return new Ability(rules);
   }
+
+  /**
+   * Given a set of {@link IdentityPermission}s, a CASL {@link Ability} is created.
+   * @param identityPermissions - a set of {@link IdentityPermission}s.
+   * @returns a CASL {@link Ability}
+   */
+  private _defineAbilitiesForIdentityPermissions(identityPermissions: IdentityPermission[]): Ability {
+    const { can, cannot, rules } = new AbilityBuilder(Ability);
+    identityPermissions.forEach((identityPermission) => {
+      const conditions = {
+        ...(identityPermission.conditions ?? {}),
+        //wildcards do not count as a condition
+        ...(identityPermission.subjectId === '*' ? {} : { subjectId: identityPermission.subjectId })
+      };
+      if (identityPermission.effect === 'ALLOW')
+        can(identityPermission.action, identityPermission.subjectType, identityPermission.fields, conditions);
+      else
```

Comment:
Do we have protection somewhere in case someone passes in something that isn't `ALLOW` or `DENY`? If not, we should probably explicitly check the value is `DENY` before assigning the `cannot` statement or do a check to make sure its a valid `effect` value

Reply:
will use a ZOD parser here to a do double check

Reply:
Resolved by adding Zod parser into `DDBDynamicAuthorizationPermissionsPlugin._transformItemToIdentityPermission`